### ISSUE TITLE
Support miui7?

### ISSUE THREAD

**Issue by exodius48** [rover12421/ShakaApktool#33]
1. is this apktool support miui 7?
2. and what is different from apktool.jar from ibotpeaches?

**Comment by rover12421** [comment]
1.I don't known either. if you have a sample can upload to here.
2.more exception handing, but don't guarantee stability.

**Comment by exodius48** [comment]
1. Wow, i can't believe this, i just tested , it works like a charm with miui 7. I can decompile & compile with no errors. But when i use ibotpeaches, i can decompile but when compiling, i have error.
Beside it, i found a bug, baksmaling classes.dex with your apktool take long time and use high cpu resource. I use windows 7 enterprise x86 , JDK & JRE 8. baksmaling classes.dex with ibotpeaches apktool is very fast.
Here as example app & framework from miui 7 http://rghost.net/6zPXkTlBg
I use this rom as example (i'm using this rom in my device) http://bigota.d.miui.com/V7.0.2.0.KHDMICI/miui_HMNoteWGlobal_V7.0.2.0.KHDMICI_06591801ab_4.4.zip
Here for the logs http://pastebin.com/cmaCqYJj

2. I think i'll use your apktool. (y)

3. I also found a bug, when i use this command apktool d -r -b ThemeManager.apk , the result like this:

[code].method public isOnlinePage()Z

    .locals 1



    iget-boolean v0, p0, Lmiui/resourcebrowser/view/ResourceOperationHandler;->mIsOnlinePage:Z



    return v0

.end method/code]
There are long gaps , 3 spaces gaps.

Sorry for my english.

**Comment by rover12421** [comment]
1. Should be performance problems caused by multi-language support. I will have time to re-design, or consider removing.

2. thx.

3. Paste code please use markdown syntax. I'll check this problem, if fix, I'll mark this issues.
```
.method public isOnlinePage()Z

.locals 1



iget-boolean v0, p0, Lmiui/resourcebrowser/view/ResourceOperationHandler;->mIsOnlinePage:Z



return v0
.end method
```

ThemeManager.apk  in MIUI 7.rar ?

**Comment by exodius48** [comment]
Yes, ThemeManager.apk inside it. :)

**Comment by rover12421** [comment]
I try is normal. You're using latest version ?

```smali
.method public isOnlinePage()Z
    .locals 1

    iget-boolean v0, p0, Lmiui/resourcebrowser/view/ResourceOperationHandler;->mIsOnlinePage:Z

    return v0
.end method
```

**Comment by exodius48** [comment]
Yes, i tried latest version from pan.baidu.com. It's weird. How did you do?

**Comment by rover12421** [comment]
You download `ShakaApktool_2.0.0-master-e30c8e8-20151120-release.jar` try again.

**Comment by exodius48** [comment]
Why i still get gaps like this

[img]http://i.imgur.com/mlz2ykp.png[/img]

Still using old command : apktool d -r -b ThemeManager.apk

**Comment by rover12421** [comment]
fix f92155ba8663f552700b7ca6865da419a44f4300

It has confirmed that the problem of windows platform.
together with performance problems have been fixed.
